iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,341 @@
+            }
+        }
+
+        private async Task RemoveAsync(string configurationId, IETagHolder eTagHolder, CancellationToken cancellationToken)
+        {
+            var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
+            {
+                {
+                    HttpStatusCode.NotFound,
+                    async responseMessage =>
+                        {
+                            string responseContent = await ExceptionHandlingHelper.GetExceptionMessageAsync(responseMessage).ConfigureAwait(false);
+                            return new ConfigurationNotFoundException(responseContent, (Exception) null);
+                        }
+                },
+                {
+                    HttpStatusCode.PreconditionFailed,
+                    async responseMessage => new PreconditionFailedException(await ExceptionHandlingHelper.GetExceptionMessageAsync(responseMessage).ConfigureAwait(false))
+                }
+            };
+            using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Delete, GetConfigurationRequestUri(configurationId), _credentialProvider);
+            HttpMessageHelper2.InsertEtag(request, eTagHolder.ETag);
+            HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+            await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.NoContent, response);
+        }
+
+        /// <summary>
+        /// Applies configuration content to an Edge device to create a deployment.
+        /// </summary>
+        /// <remarks><see cref="ConfigurationContent.ModulesContent"/> is required.
+        /// <see cref="ConfigurationContent.DeviceContent"/> is optional.
+        /// <see cref="ConfigurationContent.ModuleContent"/> is not applicable.</remarks>
+        /// <param name="deviceId">The device Id.</param>
+        /// <param name="content">The configuration.</param>
+        public virtual Task ApplyConfigurationContentOnDeviceAsync(string deviceId, ConfigurationContent content)
+        {
+            return ApplyConfigurationContentOnDeviceAsync(deviceId, content, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Applies configuration content to an Edge device.
+        /// </summary>
+        /// <remarks><see cref="ConfigurationContent.ModulesContent"/> is required.
+        /// <see cref="ConfigurationContent.DeviceContent"/> is optional.
+        /// <see cref="ConfigurationContent.ModuleContent"/> is not applicable.</remarks>
+        /// <param name="deviceId">The device Id.</param>
+        /// <param name="content">The configuration.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        public async virtual Task ApplyConfigurationContentOnDeviceAsync(string deviceId, ConfigurationContent content, CancellationToken cancellationToken)
```

Comment:
```suggestion
        public async virtual Task ApplyConfigurationContentOnDeviceAsync(string deviceId, ConfigurationContent content, CancellationToken cancellationToken = default)
```

And then delete the overload that takes no cancellation token